API/server.ts

```diff
@@ -0,0 +1,186 @@
+import * as express from 'express';
+import * as got from 'got';
+import * as fs from 'fs';
+import * as crypto from 'crypto';
+import { submit } from '../Submit';
+import { ProblemVeredict } from '../OnlineJudgeFactory/OnlineJudge';
+import { AccountAvailable, getAccountAvailable, JudgeAvailability } from './accountUtil';
+import * as appconfig from '../appconfig.json';
+
+import {
+  SupportedProgrammingLanguages,
+  ProgrammingLanguage,
+  OnlineJudge,
+  OnlineJudgesURLRegularExpression,
+  SupportedOnlineJudges,
+  ExamplesOnlineJudgeProblemURL,
+  fileTermination,
+  commentForLanguage,
+} from './utils';
+const app = express();
+const requireAuthentication =
+  appconfig['x-auth-token'] === undefined || appconfig['x-auth-token'] === '' ? false : true;
+
+app.use(express.json());
+
+async function problemExist(url: string): Promise<boolean> {
+  try {
+    const response = await got.get(url);
+    const responseURL = response.url;
+    const status = response.statusCode;
+    return responseURL === url && status >= 200 && status <= 299;
+  } catch (error) {
+    return false;
+  }
+}
+
+// Get all supported languages
+app.get('/languages', (req, res) => {
+  const authHeader = req.headers['x-auth-token'];
+  if (requireAuthentication && authHeader === undefined) return res.sendStatus(401); //Unauthorized
+  if (requireAuthentication && authHeader !== appconfig['x-auth-token']) return res.sendStatus(403); // Forbidden
```

Comment:
You can extract this code to a function and call it instead of copy-paste all the lines in each route.